src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -752,41 +813,30 @@ class NBDWatchCtx : public librbd::UpdateWatchCtx
     , io_ctx(_io_ctx)
     , image(_image)
     , size(_size)
-  { }
-
-  ~NBDWatchCtx() override {}
+  {
+    notify_thread = make_named_thread("rbd_notify", &NBDWatchCtx::notify_entry,
+                                      this);
+  }
 
-  void handle_notify() override
+  ~NBDWatchCtx() override
   {
-    librbd::image_info_t info;
-    if (image.stat(info, sizeof(info)) == 0) {
-      unsigned long new_size = info.size;
-      int ret;
+    dout(10) << __func__ << ": terminating" << dendl;
+    std::unique_lock locker{lock};
+    terminated = true;
+    cond.notify_all();
+    lock.unlock();
```

Comment:
Since locked through `locker`, better to unlock through `locker`:
```suggestion
    locker.unlock();
```